docs/src/pages/components/data-grid/rows/rows.md

```diff
@@ -15,6 +15,17 @@ The rows can be defined with the `rows` prop, which expects an array of objects.
 
 {{"demo": "pages/components/data-grid/rows/RowsGrid.js", "bg": "inline"}}
 
+> ⚠️ Each row should have a unique id.
```

Comment:
We could also mention that the "id" property doesn't need to be specified in the `columns` prop. An user wasn't aware of this: https://github.com/mui-org/material-ui-x/issues/346#issuecomment-1024584581

Reply:
Something like this ?

> ⚠️ Each row should have a unique id even when no there is no `id` column

Reply:
Interesting. Looks like the issue here is understanding `columns` - users might think that every field in their rows data is required be reflected in `columns` as well.
Maybe instead we should mention on Columns page that it's not required for `columns` to cover all the fields available in rows data?

Reply:
I copied from https://reactjs.org/docs/lists-and-keys.html#keys

```suggestion
> ⚠️ Each row should have a field that uniquely identifies the row.
```

> Maybe instead we should mention on Columns page that it's not required for columns to cover all the fields available in rows data?

It could be, but then we would need to say there that, although the "id" is required, a "id" column is not.

Reply:
@m4theushw 
> I copied from https://reactjs.org/docs/lists-and-keys.html#keys

```suggestion
> ⚠️ Each row object should have an `id` property that uniquely identifies the row.
```
Maybe this? I think it's important to mention the property name

> It could be, but then we would need to say there that, although the "id" is required, a "id" column is not.

Okay, now I'm convinced to your initial proposal to mention columns here :)

Reply:
> It could be, but then we would need to say there that, although the "id" is required, a "id" column is not.

Now I'm convinced to your initial proposal - I'll mention id column here :)

Reply:
I tried to make it as clear as I could and ended up with this:

> Note that [column definition](/components/data-grid/columns/#column-definitions) for `id` field is not required.